### PR TITLE
Store logs locally and periodically clean up

### PR DESCRIPTION
**Pull request by hsyndeniz**
### Description

- Implemented local log storage to maintain logs on the user's device for easy access and offline troubleshooting.
- Added automated log cleanup mechanism that periodically removes logs older than 1 week, optimizing storage space and ensuring a concise log history.

Screenshot 2023-07-23 at 18 50 26

### CODE REVIEW

File: src/shared/services/logging.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  },
+  async: true,
+  dateFormat: 'time',
+  printLevel: true,
+  printDate: true,
+  enabled: true,
+};
+
+export const log = logger.createLogger({
+  ...defaultConfig,
```
Comment (juanmanso) [inline src/shared/services/logging.ts:43]: so let me know if I get this right:

- This logger stores logs for the same day in the same file
- Logs start appending after the first one

Is that true? I don't see if they actually append and I'm worried that they are not (and instead they are getting overwritten over and over again.

Could you share a screenshot of how they are stored or something in an actual test?

File: src/shared/services/logging.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    const files = await RNFS.readDir(logDirectory);
+    const now = Date.now();
+    const thresholdDate = now - logRetentionPeriodInDays * 24 * 60 * 60 * 1000;
+
+    for (const file of files) {
+      if (file.isFile() && file.name.endsWith('.log')) {
+        const fileCreationTime = new Date(file.ctime ?? 0).getTime();
+        if (fileCreationTime < thresholdDate) {
+          await RNFS.unlink(file.path);
+          console.log(`Deleted log file: ${file.name}`);
```
Comment (juanmanso) [inline src/shared/services/logging.ts:72]: shouldn't we use `log` now here 🤔😅

File: src/shared/services/logging.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        }
+      }
+    }
+  } catch (error) {
+    console.log('Error deleting log files:', error);
```
Comment (juanmanso) [inline src/shared/services/logging.ts:77]: aside from the joke to use `log` here, how should we call `log` to send an error log?

Can you use this case as an example to show how to console.error here?

File: src/shared/services/logging.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  )
+    .then(contents => {
+      return contents;
+    })
+    .catch(err => {
+      console.log(err);
+    });
+};
+
+export const deleteOldLogFiles = async (): Promise<void> => {
```
Comment (juanmanso) [inline src/shared/services/logging.ts:61]: where are we using the function to run this continuously? Does it run on every render or should it run on startup?

I think the latter, but I don't see it in this PR. Maybe a commit is missing or something?

**Review by juanmanso — COMMENTED** [review]
Left some comments!

File: src/shared/services/logging.ts (hunk 2 above)
Comment (hsyndeniz) [reply src/shared/services/logging.ts]: Sure! Why not? 😄

File: src/shared/services/logging.ts (hunk 3 above)
Comment (hsyndeniz) [reply src/shared/services/logging.ts]: You can use it just like the console.log
log.debug("This is a Debug log");
log.info("This is an Info log");
log.warn("This is a Warning log");
log.error("This is an Error log");

File: src/shared/services/logging.ts (hunk 4 above)
Comment (hsyndeniz) [reply src/shared/services/logging.ts:61]: We call it in App.tsx now.

**Review by juanmanso — APPROVED** [review]
LGTM. Left some comments unresolved to use them as pseudo documentation 😅

**Pull request merged**